mismi-s3/test/Test/Mismi/S3.hs

```diff
@@ -66,3 +75,24 @@ withToken t f = do
   u <- liftIO $ T.pack . U.toString <$> U.nextRandom
   let a = Address b (Key . T.intercalate "/" $ ["mismi", u, unToken t])
   awsBracket_ (pure ()) (listRecursively a >>= mapM_ delete >> delete a) (f a)
+
+
+newAddress :: Token -> AWS Address
+newAddress t = do
+  b <- liftIO testBucket
+  u <- liftIO $ T.pack . U.toString <$> U.nextRandom
+  let a = Address b (Key . T.intercalate "/" $ ["mismi", u, unToken t])
+  e <- ask
+  void $ register (eitherT throwM pure . runAWS e $ listRecursively a >>= mapM_ delete >> delete a)
+  pure $ a
+
+newFilePath :: Token -> AWS FilePath
+newFilePath t = do
+  p <- liftIO $ do
+    d <- getTemporaryDirectory
```

Comment:
Was this what it was doing before? If so, just leave, but I would like to make the builds a bit more self contained at some point :/